src/main/java/seedu/address/model/Model.java

```diff
@@ -165,6 +165,15 @@ public interface Model {
      */
     boolean hasLecture(ModuleCode moduleCode, LectureName lectureName);
 
+    /**
+     * Returns true if a {@code Module} has a lecture with name {@code lecture}.
+     *
+     * @param module The module to check if it contains the lecture.
+     * @param lecture The lecture to check if exist.
+     * @return True if a lecture with the same name as {@code lecture} exists in {@code module}. Otherwise, false.
+     */
+    boolean hasLecture(ReadOnlyModule module, LectureName lecture);
+
```

Comment:
Don't think this is necessary. Consider using the `hasLecture(ModuleCode, LectureName)` method.